Get rid of 'rash' gem.

barthez opened the pull request:
Closes #1

barthez committed:
Get rid of 'rash' gem.

masone commented on lib/twofishes/result.rb:
```diff
@@ -16,41 +12,51 @@ def self.from_response(hash)
 
-    def name
-      feature.name
+    def what
```
this changes the behavior compared to the implementation before (name is gone). also introduces a breaking change. accident or on purpose?

masone commented on lib/twofishes/result.rb:
```diff
@@ -16,41 +12,51 @@ def self.from_response(hash)
 
-    def display_name
-      feature.display_name
+    def where
```
this changes the behavior compared to the implementation before (display_name is gone). also introduces a breaking change. accident or on purpose?

masone commented on lib/twofishes/result.rb:
```diff
@@ -16,41 +12,51 @@ def self.from_response(hash)
 
+    private
+
+    def derubyfy_key(key)
```
what about changing the naming to `camelize`? that's what it is called in rails.

masone commented on lib/twofishes/result.rb:
```diff
@@ -16,41 +12,51 @@ def self.from_response(hash)
+    private
+
+    def derubyfy_key(key)
+      key.to_s.gsub(/_[a-z]/) { |m| m[1].upcase }
```
i think it's currently not required for twofishes, but should we support method names with other characters as a-z in it?

more generic: 

```
key.split('_').collect(&:capitalize).join
```

miharekar replied:
Let's just use the `#underscore` from ActiveSupport::Inflector. We can add the active_support gem to the gemfile without requiring and then explicitly `require 'active_support/core_ext/string/inflections'`. http://edgeguides.rubyonrails.org/active_support_core_extensions.html#underscore

> Active Support has been carefully revised so that cherry-picking a file loads only strictly needed dependencies, if any.

masone commented on test/twofishes/result_test.rb:
```diff
@@ -37,7 +37,7 @@
     end
 
-    it "should return display_name" do
+    it "should return display_name (rubyfied)" do
```
i don't understand what the rubyfied is here for? did you want to separate tests for `result.displayName`and `display_name` instead?

barthez committed:
Introduce ResultFields instead of rash

Preserve backward compability

barthez commented:
`missing_method` call should not omit calling super. There `missing_method` always respond with some value (mostly `nil`), but it should not be like this:

```
Result.respond_to?(:foobar) # => false
Result.foobar # => nil
```

This is more logical behavior:

```
Result.respond_to?(:foobar) # => false
Result.foobar # => raise NoMethodError
```

It will introduce breaking changes, so if you agree to do so, you also need to update minor version number.

miharekar commented:
BTW Breaking changes call for major version number update, not minor.

Commit:
Use Forwardable module

- remove missing method rescue from Result (breaking change)
- raise NoMethodError on missing attribute in ResultFields

miharekar commented on lib/twofishes/result.rb:
```diff
@@ -16,17 +14,12 @@ def self.from_response(hash)
-    def country_code
-      feature.cc
-    end
+    def_delegator :@data, :what
```
`def_delegators :@data, :what, :where, :feature`?

barthez replied:
I make it that way first, but i looked like:

```
def_delegators :@data, :what, :where, :feature
def_delegators :feature, :name, :display_name
def_delegator :feature, :cc, :country_code
```

What could be confusing, that's why I decide to write each delegation in separated line.

miharekar replied:
```
def_delegators :@data, :what, :where, :feature
def_delegators :feature, :name, :display_name

def_delegator :feature, :cc, :country_code
```

That's pretty readable I guess. But yeah, I understand now why you did it your way. As you wish, both is fine with me.

masone commented on Gemfile:
```diff
@@ -2,3 +2,5 @@ source 'https://rubygems.org'
 # Specify your gem's dependencies in twofishes.gemspec
 gemspec
+
+gem 'activesupport', :require => false
```
i think this definition can be removed. after adding it to the gemspec, it still needs to be required manually.

miharekar replied:
https://github.com/mrfoto/twofishes-ruby/commit/a169fbe58d793f068fabb6b13cfd53c8432b1659

miharekar replied:
#4 already includes these commits :wink:

masone commented on lib/twofishes/result_fields.rb:
```diff
@@ -0,0 +1,37 @@
+    alias_method :to_hash, :to_h
+
+    def [](key)
+      wrap_value @hash[key.to_s.camelize(:lower)]
```
shall we extract `key.to_s.camelize(:lower)` to a private method?

miharekar replied:
Again, stop looking at this, look at #4 :smile: Or have the day off, like the doctor ordered

masone replied:
you a doctor?

miharekar replied:
:ambulance:

masone commented:
closing in favor of #4. Thanks, this looks very clean!
